File: dotnet/ArqLibrarianClassic/ArqLibrarianClassic/Library/BorrowingManager.cs
using ArqLibrarianClassic.Library;

namespace ArqLibrarianClassic
{
    public class BorrowingManager
    {
        private readonly UserDao userDao;
        private readonly BooksDao bookDao;
        private readonly BorrowingDao borrowingDao;

        public BorrowingManager(UserDao userDao, BooksDao bookDao, BorrowingDao borrowingDao)
        {
            this.userDao = userDao;
            this.bookDao = bookDao;
            this.borrowingDao = borrowingDao;
        }

        public void Borrow(long userId, long bookId)
        {
            var user = userDao.FindById(userId);
            var book = bookDao.FindById(bookId);

            borrowingDao.Insert(new Borrowing(user, book));
        }

        public bool Borrowed(long bookId)
        {
            return borrowingDao.FindByBookId(bookId) != null;
        }
    }
}

File: java/ArqLibraryClassic/ArqLibraryLayered/src/bnsit/ms/arq/library/BorrowingManager.java
package bnsit.ms.arq.library;

public class BorrowingManager {
    private UserDao userDao;
    private BooksDao booksDao;
    private BorrowingDao borrowingDao;

    public BorrowingManager(UserDao userDao, BooksDao booksDao, BorrowingDao borrowingDao) {
        this.userDao = userDao;
        this.booksDao = booksDao;
        this.borrowingDao = borrowingDao;
    }

    public void borrow(long userId, long bookId) {
        User user = userDao.findById(userId);
        Book book = booksDao.findById(bookId);

        borrowingDao.insert(new Borrowing(user, book));
    }

    public boolean borrowed(long bookId) {
        return borrowingDao.findByBookId(bookId) != null;
    }
}
